src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs

```diff
@@ -371,9 +371,38 @@ private void UpdateInputScope(InputScope inputScope)
 
 		partial void OnTextWrappingChangedPartial(DependencyPropertyChangedEventArgs e)
 		{
-			//TODO : see bug #8178
+			if (e.NewValue is TextWrapping textWrapping && !string.IsNullOrEmpty(Text) &&
+				(textWrapping is TextWrapping.Wrap || textWrapping is TextWrapping.WrapWholeWords))
+			{
+				if (_textBoxView != null)
+				{
+					//store  the value configured on the wrapper _textBoxView to restore if user changes from Wrap to UnWrap
+					_isSingleLine = _textBoxView.IsSingleLine;
+
+					//just change if IsSingleLine equals true, to avoid invoke the propertychange in the wrong way
+					if (_textBoxView.IsSingleLine)
+					{
+						//this must be setted to false to Wrap the text
+						_textBoxView?.SetSingleLine(false);
+					}
+				}
+			}
+			else
+			{
+				//just check if the "user value" was stored 
+				if (_isSingleLine.HasValue)
+				{
+					//return the value configured by user
+					_textBoxView?.SetSingleLine(_isSingleLine.Value);
+
+					//reset the value
+					_isSingleLine = null;
+				}
+			}
 		}
 
+		private bool? _isSingleLine;
```

Comment:
Consider moving this line to the existing fields at the top:

https://github.com/unoplatform/uno/blob/6aa0f7003820bb5b0eaafcce6902943e9df0247c/src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs#L33-L37